packages/framework-provider-aws/src/index.ts

```diff
@@ -57,63 +9,66 @@ function loadInfrastructurePackage(packageName: string): HasInfrastructure {
  */
 export const Provider = (rockets?: RocketDescriptor[]): ProviderLibrary => {
-  return {
-    // ProviderEventsLibrary
-    events: {
-      rawToEnvelopes: rawEventsToEnvelopes,
-      forEntitySince: readEntityEventsSince.bind(null, dynamoDB),
-      latestEntitySnapshot: readEntityLatestSnapshot.bind(null, dynamoDB),
-      search: searchEvents.bind(null, dynamoDB),
-      store: storeEvents.bind(null, dynamoDB),
-    },
-    // ProviderReadModelsLibrary
-    readModels: {
-      rawToEnvelopes: rawReadModelEventsToEnvelopes,
-      fetch: fetchReadModel.bind(null, dynamoDB),
-      search: searchReadModel.bind(null, dynamoDB),
-      store: storeReadModel.bind(null, dynamoDB),
-      delete: deleteReadModel.bind(null, dynamoDB),
-      subscribe: subscribeToReadModel.bind(null, dynamoDB),
-      fetchSubscriptions: fetchSubscriptions.bind(null, dynamoDB),
-      deleteSubscription: deleteSubscription.bind(null, dynamoDB),
-      deleteAllSubscriptions: deleteAllSubscriptions.bind(null, dynamoDB),
-    },
-    // ProviderGraphQLLibrary
-    graphQL: {
-      rawToEnvelope: rawGraphQLRequestToEnvelope,
-      handleResult: requestSucceeded,
-    },
-    // ProviderAPIHandling
-    api: {
-      requestSucceeded,
-      requestFailed,
-    },
-    connections: {
-      storeData: storeConnectionData.bind(null, dynamoDB),
-      fetchData: fetchConnectionData.bind(null, dynamoDB),
-      deleteData: deleteConnectionData.bind(null, dynamoDB),
-      sendMessage: sendMessageToConnection,
-    },
-    // ScheduledCommandsLibrary
-    scheduled: {
-      rawToEnvelope: rawScheduledInputToEnvelope,
-    },
-    // ProviderInfrastructureGetter
-    infrastructure: () => {
-      const infrastructurePackageName = require('../package.json').name + '-infrastructure'
-      let infrastructure: HasInfrastructure | undefined
-      try {
-        infrastructure = loadInfrastructurePackage(infrastructurePackageName)
-      } catch (e) {
-        throw new Error(
-          `The AWS infrastructure package could not be loaded. The following error was thrown: ${e.message}. Please ensure that one of the following actions has been done:\n` +
-            `  - It has been specified in your "devDependencies" section of your "package.json" file. You can do so by running 'npm install --save-dev ${infrastructurePackageName}'\n` +
-            `  - Or it has been installed globally. You can do so by running 'npm install -g ${infrastructurePackageName}'`
-        )
-      }
-
-      return infrastructure.Infrastructure(rockets)
-    },
+  try {
+    require('aws-sdk')
+    const { Provider } = require('./setup.ts')
+    return Provider(rockets)
+  } catch (e) {
+    return {
+      // ProviderEventsLibrary
+      events: {
+        rawToEnvelopes: undefined as any,
+        forEntitySince: undefined as any,
+        latestEntitySnapshot: undefined as any,
+        store: undefined as any,
+        search: undefined as any,
+      },
```

Comment:
Sweet! I'm still wondering if it would be worth moving this `empty provider` to a generic package 🤔  

It doesn't have to be in this PR, but it would be worth having access to it from any package.